js-user-registration-service/src/test/java/com/jumia/warmup/jsuserregistrationservice/rest/UserRegistrationRestTest.java

```diff
@@ -0,0 +1,259 @@
+
+        userDTO.getAccountInformationDTO().setUserName("");
+
+        mvc.perform(post(API_USERS)
+            .contentType(APPLICATION_JSON)
+            .content(asJsonString(userDTO))
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest())
+            .andExpect(content().string(containsString(Constants.ErrorMessages.EMPTY_USER_NAME)));
+    }
+
+    @Test
+    public void registerUser_expectBadeRequest_userName_lessThan4Chars() throws Exception {
+
+        userDTO.getAccountInformationDTO().setUserName("aaa");
+
+        mvc.perform(post(API_USERS)
+            .contentType(APPLICATION_JSON)
+            .content(asJsonString(userDTO))
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest())
+            .andExpect(content().string(containsString(Constants.ErrorMessages.INVALID_USER_NAME)));
+    }
+
+    @Test
+    public void registerUser_expectBadeRequest_userName_greaterThan10Chars() throws Exception {
+
+        userDTO.getAccountInformationDTO().setUserName("asdhyuikjhdthtr");
+
+        mvc.perform(post(API_USERS)
+            .contentType(APPLICATION_JSON)
+            .content(asJsonString(userDTO))
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest())
+            .andExpect(content().string(containsString(Constants.ErrorMessages.INVALID_USER_NAME)));
+    }
+
+    @Test
+    public void registerUser_expectBadeRequest_password_null() throws Exception {
+
+        userDTO.getAccountInformationDTO().setPassword(null);
+
+        mvc.perform(post(API_USERS)
+            .contentType(APPLICATION_JSON)
+            .content(asJsonString(userDTO))
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest())
+            .andExpect(content().string(containsString(Constants.ErrorMessages.EMPTY_PASSWORD)));
+    }
+
+
+    @Test
+    public void registerUser_expectBadeRequest_password_empty() throws Exception {
+
+        userDTO.getAccountInformationDTO().setPassword("");
+
+        mvc.perform(post(API_USERS)
+            .contentType(APPLICATION_JSON)
+            .content(asJsonString(userDTO))
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest())
+            .andExpect(content().string(containsString(Constants.ErrorMessages.EMPTY_PASSWORD)));
+    }
+
+    @Test
+    public void registerUser_expectBadeRequest_password_noCapitalCharacters() throws Exception {
+
+        userDTO.getAccountInformationDTO().setPassword("somepassword12@#");
+
+        mvc.perform(post(API_USERS)
+            .contentType(APPLICATION_JSON)
+            .content(asJsonString(userDTO))
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest())
+            .andExpect(content().string(containsString(ErrorMessages.INVALID_PASSWORD)));
+    }
+
+//    @Test
```

Comment:
Commented code should be removed if not used :)

Reply:
Done.